wire.go

```diff
@@ -226,12 +257,37 @@ func (srv *Server) serve(ctx context.Context, conn net.Conn) error {
 
 // Close gracefully closes the underlaying Postgres server.
 func (srv *Server) Close() error {
-	if srv.closing.Load() {
-		return nil
+	if !srv.closing.CompareAndSwap(false, true) {
+		return nil // Already closing
 	}
 
-	srv.closing.Store(true)
 	close(srv.closer)
-	srv.wg.Wait()
-	return nil
+	
+	// Wait for server to start before waiting for graceful shutdown
+	select {
+	case <-srv.started:
+		// Server has started, proceed with graceful shutdown
+	default:
+		// Server hasn't started yet, no need to wait
+		return nil
+	}
+	
+	// Graceful shutdown with 1-second timeout
+	done := make(chan struct{})
+	go func() {
+		srv.wg.Wait()
+		close(done)
+	}()
+	
+	select {
+	case <-done:
+		// All connections closed gracefully
+		return nil
+	case <-time.After(1 * time.Second):
```

Comment:
I think we should stick to the stdlib convention. See the docs for Close() vs Shutdown():

https://pkg.go.dev/net/http#Server.Close
https://pkg.go.dev/net/http#Server.Shutdown

Because Shutdown() accepts a ctx argument that allows the integrator to set the timeout. The way this code is structured there is no way to change it.